packages/flame/lib/src/components/base_component.dart

```diff
@@ -236,4 +201,12 @@ abstract class BaseComponent extends Component {
   Vector2 eventPosition(PositionInfo info) {
     return isHud ? info.eventPosition.widget : info.eventPosition.game;
   }
+
+  ComponentSet createComponentSet() {
+    final components = ComponentSet.createDefault(prepare);
+    if (this is HasGameRef) {
+      components.register<HasGameRef>();
```

Comment:
this allows HasGameRef to do `query<HasGameRef>()`, which should improve performance.
also I make all ordered sets QOS by default. so no more complications into whether we can use QOS's features or not